modules/dnsmasq_dhcp/autodetection.go

```diff
@@ -46,18 +60,128 @@ func (d *DnsmasqDHCP) findDHCPRanges() ([]string, error) {
 	return ranges, nil
 }
 
-func findConfigurations(confDir string) ([]string, error) {
+// findConfigs recursively finds and reads configuration files respecting
+// conf-file and conf-dir directives.
+// findConfigs is tolerant to IO errors and finds as maximum config
+// files as possible, therefore if an error occurrs during scanning process,
+// it will be just logged with warning severity
+func (d *DnsmasqDHCP) findConfigs(confPath string) []string {
+	config, err := os.Open(confPath)
+	if err != nil {
+		d.Warningf("error during configuration file %q reading: %v", confPath, err)
+		return nil
+	}
+
+	defer config.Close()
+
+	var (
+		includeFiles []string
+		includeDirs  []configDir
+	)
+
+	scanner := bufio.NewScanner(config)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if path, ok := getConfValue(line, "conf-file"); ok {
+			includeFiles = append(includeFiles, path)
+			continue
+		}
+
+		if path, ok := getConfValue(line, "conf-dir"); ok {
+			args := strings.Split(path, ",")
+
+			dir := configDir{
+				path: args[0],
+			}
+
+			for _, arg := range args[1:] {
+				arg := strings.TrimSpace(arg)
+				// dnsmasq treats suffixes with asterisk as "to include" and without
+				// asterisk as "to exclude"
+				if strings.HasPrefix(arg, "*") {
+					dir.includeSuffix = append(dir.includeSuffix, arg[1:])
+				} else {
+					dir.excludeSuffix = append(dir.excludeSuffix, arg)
+				}
+			}
+
+			includeDirs = append(includeDirs, dir)
+
+			continue
+		}
+	}
+
+	for _, dir := range includeDirs {
+		dirFiles, err := readConfDir(dir.path, dir.includeSuffix, dir.excludeSuffix)
```

Comment:
why not `readConfDir(dir)`

i see 

```go
type configDir struct
```

and func name `readConfDir`, i think it is ok to pass `configDir` struct there

Reply:
That's good point about passing structure, I thought about it and decided to abstract the function of struct to show that the func depends only on specific input fields and not on whole struct. But since struct configDir is not going to be changed heavily, we can pass it entirely.

Agreed, the func name is not the best here because it not only reads, but also filters, thought about findConfigsInDir (don't like that `In` in name), scanConfigDir (but scan sounds ambiguous), readAndFilterConfigDir (too long, but sounds idiomatic for Go, like http.ListenAndServe). What do you think?

Reply:
> readAndFilter

i think we can name it `find` and make it a method of configDir

> findConfigFiles

what do you think?

Reply:
:+1: